srml/staking/src/lib.rs

```diff
@@ -1502,6 +1524,13 @@ impl <T: Trait> OnOffenceHandler<T::AccountId, session::historical::Identificati
 				continue
 			}
 
+			// Auto deselect validator on any offence and force a new era if they haven't previously
+			// been deselected.
+			if <Validators<T>>::exists(stash) {
```

Comment:
I think it would be better to move it after the `amount` check. We don't really slash anything if the amount is zero. Now we will deselect the validator even if they are not really slashed.

Reply:
That's exactly the point.

> Force offending validators (*even those offenders whose slash was zero*) to take manual action to avoid deselection